packages/block-editor/src/hooks/color.js

```diff
@@ -334,6 +335,48 @@ export function ColorEdit( props ) {
 	);
 }
 
+/**
+ * This adds inline styles for color palette colors.
+ * Ideally, this is not needed and themes should load their palettes on the editor.
+ *
+ * @param  {Function} BlockListBlock Original component
+ * @return {Function}                Wrapped component
+ */
+export const withColorPaletteStyles = createHigherOrderComponent(
+	( BlockListBlock ) => ( props ) => {
+		const { name, attributes } = props;
+		const { backgroundColor, textColor } = attributes;
+		const { colors } = useSelect( ( select ) => {
+			return select( 'core/block-editor' ).getSettings();
```

Comment:
Why are we returning all settings here? Shouldn't we only subscribe to `colors` changes?

Reply:
Good remark, not sure it has any impact though.